src/protocols/types/datastore.js

```diff
@@ -788,9 +788,21 @@ class DataStorePrepareUpdateParam extends NEXTypes.Structure {
 			nexVersion = stream.connection.title.nex_version;
 		}
 
-		this.dataId = stream.readInt64LE();
+		if (nexVersion.major >= 3) {
+			this.dataId = stream.readUInt64LE();
+
+			// * Hack to get the proper type when encoding to JSON
+			this.dataIdType = 'uint64';
```

Comment:
since `dataIdType` is not really part of the type, this should probably be made into a private field like `#dataIdType` to differentiate it at a glance